pkg/controller/controller/group_controller.go

```diff
@@ -301,3 +322,53 @@ func isPodBelongsToGroup(group *crd.Group, pod *api_v1.Pod) bool {
 	}
 	return false
 }
+
+func getPodVolumesMap(pod api_v1.Pod) map[string]*crd.PersistentVolumeClaim {
+	podVolumes := map[string]*crd.PersistentVolumeClaim{}
+	for j := 0; j < len(pod.Spec.Volumes); j++ {
+		vol := pod.Spec.Volumes[j]
+		if vol.PersistentVolumeClaim != nil {
+			claimName := vol.PersistentVolumeClaim.ClaimName
+			podVolumes[claimName] = collectPersistentVolumeClaim(claimName)
+			podVolumes[claimName].BoundTimes = append(podVolumes[claimName].BoundTimes, pod.GetCreationTimestamp())
+		}
+	}
+
+	return podVolumes
+}
+
+func collectPersistentVolumeClaim(claimName string) *crd.PersistentVolumeClaim {
+	pvc, err := ClientSetInstance.CoreV1().PersistentVolumeClaims("").Get(claimName, meta_v1.GetOptions{})
+	if errors.IsNotFound(err) {
+		log.Printf("Persistent Volume Claim %s not found\n", claimName)
+		return nil
+	} else if statusError, isStatus := err.(*errors.StatusError); isStatus {
+		log.Printf("Error getting persistence volume Claim %s : %v\n", claimName, statusError.ErrStatus.Message)
+		return nil
+	} else if err != nil {
+		panic(err.Error())
+	} else {
+		request := pvc.Spec.Resources.Requests["storage"].DeepCopy()
+		capacity := pvc.Status.Capacity["storage"].DeepCopy()
+
+		return &crd.PersistentVolumeClaim{
+			Name:                pvc.GetObjectMeta().GetName(),
+			VolumeName:          pvc.Spec.VolumeName,
+			RequestSizeInGB:     []float64{utils.BytesToGB(request.Value())},
+			CapacityAllotedInGB: []float64{utils.BytesToGB(capacity.Value())},
+			BoundTimes:          []meta_v1.Time{},
+			UnboundTimes:        []meta_v1.Time{},
+		}
+	}
```

Comment:
Use of too many `if-else` constructs is generally not a good practise. 

I would recommend breaking this into something like: 

``` go
if err != nil {
		if errors.IsNotFound(err) {
			log.Printf("Persistent Volume Claim %s not found\n", claimName)
		}

		if statusError, isStatus := err.(*errors.StatusError); isStatus {
			log.Printf("Error getting persistence volume Claim %s : %v\n", claimName, statusError.ErrStatus.Message)
		}
		return nil
	}

	request := pvc.Spec.Resources.Requests["storage"].DeepCopy()
	capacity := pvc.Status.Capacity["storage"].DeepCopy()

	return &crd.PersistentVolumeClaim{
		Name:                pvc.GetObjectMeta().GetName(),
		VolumeName:          pvc.Spec.VolumeName,
		RequestSizeInGB:     []float64{utils.BytesToGB(request.Value())},
		CapacityAllotedInGB: []float64{utils.BytesToGB(capacity.Value())},
		BoundTimes:          []meta_v1.Time{},
		UnboundTimes:        []meta_v1.Time{},
	}
``` 

Probably, even abstracting the error handling into a separate method something like `checkAndLogErr()`.

Reply:
Done.